packages/compass-collection/src/stores/collection-tab.ts

```diff
@@ -100,15 +76,15 @@ export function activatePlugin(
   );
 
   on(localAppRegistry, 'open-create-index-modal', () => {
-    store.dispatch(selectTab('Indexes'));
+    store.dispatch(selectTab(CollectionSubtabs.Indexes));
   });
 
   on(localAppRegistry, 'open-create-search-index-modal', () => {
-    store.dispatch(selectTab('Indexes'));
+    store.dispatch(selectTab(CollectionSubtabs.Indexes));
   });
 
   on(localAppRegistry, 'generate-aggregation-from-query', () => {
-    store.dispatch(selectTab('Aggregations'));
+    store.dispatch(selectTab(CollectionSubtabs.Aggregations));
   });
```

Comment:
Now that this state lives in workspaces plugin, we need to remove this logic from this store

Reply:
In this case, these events will only be applicable to the active namespace, right?

Reply:
They should stay attached to the tab local app registry scope. If we're moving the state up, this generally means that we should just set up the subscription when the tab is first created

Reply:
Added in [cec4882](https://github.com/mongodb-js/compass/pull/5568/commits/cec4882641fbe89770c4e5565f41b7d05a729022)